### PR TITLE
Provide more info via CM_Http_ClientDevice

### PR DESCRIPTION
**Pull request by zazabe**
Rename `CM_Http_ClientDevice` into `CM_Http_ClientInfo` and retrieve more info about the client (remote IP address for now).
Pretty useful to have access to such info via the `CM_Frontend_Environment`, ie. from a component.

cc @njam

### CODE REVIEW

**Comment by zazabe** [comment]
@fvovan @njam wdyt about this change?

File: library/CM/Http/ClientInfo.php [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
+        $this->_parser = new Agent($headerList);
         $this->_headerList = $headerList;
+        $this->_server = $request->getServer();
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getIp() {
+        return isset($this->_server['remote_addr']) ? $this->_server['remote_addr'] : null;
```
Comment (njam) [inline library/CM/Http/ClientInfo.php]: Use `Request::getIp()` instead?

File: library/CM/Frontend/Environment.php [hunk 2]
```diff
@@ -200,16 +200,16 @@ public function getCurrency() {
...
      */
-    public function setClientDevice(CM_Http_ClientDevice $clientDevice = null) {
-        $this->_clientDevice = $clientDevice;
+    public function setClientInfo(CM_Http_ClientInfo $clientInfo = null) {
+        $this->_clientInfo = $clientInfo;
     }
 
     /**
-     * @return CM_Http_ClientDevice
+     * @return CM_Http_ClientInfo
```
Comment (njam) [inline library/CM/Frontend/Environment.php]: Docu: can be null

**Comment by fvovan** [comment]
btw it seems it's breaking.
It could be good idea to check all usages in the app.

**Comment by njam** [comment]
Ah, AC :‑O

I don't like the name "client info" too much, it's very generic.
In the context of "environment" I think "client device" would still be clearer. The IP address is an attribute of the client's device. wdyt?

File: library/CM/Http/ClientInfo.php (hunk 1 above)
Comment (zazabe) [inline library/CM/Http/ClientInfo.php]: but it's a `ip2long()` representation, not something like `x.x.x.x`

**Comment by zazabe** [comment]
ok, let's keep the previous naming, would avoid conflict in the app too ;)

**Comment by zazabe** [comment]
@fvovan please check again

File: library/CM/Http/ClientInfo.php (hunk 1 above)
Comment (njam) [inline library/CM/Http/ClientInfo.php]: I just meant to keep the extraction from SERVER in a single place (Request).
We could add another function to Request that returns the IP address with dots? It's anyway a bit strange that `getIp()` returns a number..

**Comment by zazabe** [comment]
@fvovan sorry, i did some changes to use `Request::getIp()` as @njam suggested. it's ready of a review now :)

File: library/CM/Http/ClientInfo.php (hunk 1 above)
Comment (zazabe) [inline library/CM/Http/ClientInfo.php]: I did it with `Request::getIp($dotNotation = null)` to not break the app, is it acceptable?

**Pull request merged**